pkg/registry/core/pod/storage/eviction.go

```diff
@@ -226,16 +226,18 @@ func (r *EvictionREST) checkAndDecrement(namespace string, podName string, pdb p
 		return err
 	}
 
-	pdb.Status.PodDisruptionsAllowed--
+	if pdb.Status.DisruptedPods == nil {
+		pdb.Status.DisruptedPods = make(map[string]metav1.Time)
+	}
+
+	if _, ok := pdb.Status.DisruptedPods[podName]; ok {
```

Comment:
We only want to decrement when we are seeing a pod that isn't already in the `DisruptedPods` map right? This only decrements for pods that are already in the map. Maybe add a test for this.

Reply:
@mortent 
Please take another look.

Reply:
The existing test uses REST API to simulate various scenarios.

Still need to figure out how to make assertion w.r.t. the decrement.